src/Polynomials.jl

```diff
@@ -363,6 +365,7 @@ p(4)   # 57 = 1 + 2*4 + 3*4^2
 """
 function polyval{T,S}(p::Poly{T}, x::S)
     R = promote_type(T,S)
+    T<:AbstractFloat && any(isnan(p.a)) && convert(R, NaN)
```

Comment:
There is a problem in `T<:AbstractFloat && any(isnan(p.a))` type of check. In Julia, `?isnan` says the function checks whether a given floating point number is not a number. However, `isnan(5)` is well defined. So is `isnan(1. + 5im)`. If I skip `T<:AbstractFloat`, `p(Mod2(false)) == p[0]` test gives error, saying that `isnan` is not defined for `Mod2`.

How should we proceed?